OpenMTRDemo/MainForm.cs

```diff
@@ -7,38 +7,156 @@
 using System.Text;
 using System.Threading.Tasks;
 using System.Windows.Forms;
+using OpenCvSharp;
 using OpenMTR;
+using System.Text.RegularExpressions;
+using System.IO;
 
 namespace OpenMTRDemo
 {
     public partial class MainForm : Form
     {
-        OpenFileDialog FileBrowser;
-        Meter meter;
+        OpenFileDialog OpenBrowser;
+        SaveFileDialog SaveBrowser;
+        OpenMTR.Meter meter;
+
         public MainForm()
         {
-            FileBrowser = new OpenFileDialog();
-            FileBrowser.Filter = "Image Files|*.bmp;*.jpeg;*.jpg;*.gif";
+            OpenBrowser = new OpenFileDialog();
+            SaveBrowser = new SaveFileDialog();
+            OpenBrowser.Filter = "Image Files|*.bmp;*.gif;*.jpg;*.jpeg;*.png";
+            SaveBrowser.Filter = "JPG|*.jpg";
             InitializeComponent();
         }
 
-        private void loadToolStripMenuItem_Click(object sender, EventArgs e)
+        private void openToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            if (OpenBrowser.ShowDialog() == DialogResult.OK)
+            {
+                SetMeter(OpenBrowser.FileName);
+                MetaDataTextBox.Text = meter.MeterRead + "";
+                SetDisableableControls(true);
+                WidthTextBox.Text = meter.SourceImage.Width + " px";
+                HeightTextBox.Text = meter.SourceImage.Height + " px";
+            }
+        }
+
+        private void SetMeter(string path)
         {
-            if (FileBrowser.ShowDialog() == DialogResult.OK)
+            string metaPath = Regex.Replace(Path.GetFullPath(path), @"\.(bmp|gif|jpe?g|png)$", ".txt", RegexOptions.IgnoreCase);
+            int metaData;
+            try
             {
-                meter = ReadData.GetMeter(FileBrowser.FileName);
-                Render();
+                metaData = Int32.Parse(File.ReadAllText(metaPath));
             }
+            catch
+            {
+                metaData = -1;
+            }
+            meter = new Meter(path, new Mat(path), new Mat(path), metaData);
+            Render();
         }
```

Comment:
I don't think the above method is pertinent in the demo application at the present time. Stash this away in another PR somewhere and remove it from this PR.